library/x509.c

```diff
@@ -775,17 +775,24 @@ int mbedtls_x509_dn_gets( char *buf, size_t size, const mbedtls_x509_name *dn )
             ret = mbedtls_snprintf( p, n, "\?\?=" );
         MBEDTLS_X509_SAFE_SNPRINTF;
 
-        for( i = 0; i < name->val.len; i++ )
+        for( i = 0, j = 0; i < name->val.len; i++, j++ )
         {
-            if( i >= sizeof( s ) - 1 )
+            if( j >= sizeof( s ) - 1 )
```

Comment:
I think this should be updated as per `development`. There's an argument that a more correct approach (erroring here) but that changes the behaviour on existing data (which this would) is okay on `development` but should be left the same on LTS, but I'd prefer that the code was the same, for ease of future maintenance.